zerver/lib/notifications.py

```diff
@@ -142,9 +144,15 @@ def build_sender_payload(message):
                 'content': [build_message_payload(message)]}
 
     def message_header(user_profile, message):
-        # type: (UserProfile, Message) -> Dict[str, str]
+        # type: (UserProfile, Message) -> Dict[str, text_type]
         disp_recipient = get_display_recipient(message.recipient)
-        if message.recipient.type == Recipient.PERSONAL:
+        if isinstance(disp_recipient, text_type):
```

Comment:
this would be clearer if written as `if message.recipient.type == Recipient.STREAM` followed by an `assert isinstance(disp_recipient, text_type)`.